source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -287,16 +287,18 @@ uint16_t InstanceImpl::ThreadLocalPool::shardSize() {
   }
 
   Common::Redis::RespValue request;
-  for (uint16_t size = 0;; size++) {
+  std::set<Upstream::HostConstSharedPtr> uniqueHosts;
+  for (uint16_t size = 0; size < Envoy::Extensions::Clusters::Redis::MaxSlot; size++) {
     Clusters::Redis::RedisSpecifyShardContextImpl lb_context(
         size, request, Common::Redis::Client::ReadPolicy::Primary);
     Upstream::HostConstSharedPtr host = Upstream::LoadBalancer::onlyAllowSynchronousHostSelection(
         cluster_->loadBalancer().chooseHost(&lb_context));
     if (!host) {
       return size;
     }
+    uniqueHosts.insert(host);
```

Comment:
The compiler may do this automatically, but just to be explicit you should move the shared ptr into the set:
```suggestion
    uniqueHosts.insert(std::move(host));
```

Reply:
Just implemented this change as well, thank you for the feedback!